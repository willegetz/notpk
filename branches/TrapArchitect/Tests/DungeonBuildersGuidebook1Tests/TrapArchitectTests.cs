using System.Collections.Generic;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using DungeonBuildersGuidebook1;
using RpgTools.Dice;
using System;
using NUnit.Framework;

namespace DungeonBuildersGuidebook1Tests
{
	[TestFixture]
	[UseReporter(typeof(DiffReporter))]
	public class TrapArchitectTests
	{
		// The user should also be allowed to pick their Trap Basic
		//		Some form of display to pick from

		[Test]
		public void SpeceficTrapTest()
		{
			DiceCup.SetRandom(new Random(4));
			var architect = new TrapArchitect();

			var trapBase = architect.GetSpecificTrapBase(15);
			var trapEffect = architect.GetSpecificTrapEffect(33);
			var trapDamage = architect.GetSpecificTrapDamage(8);
			var trap = new Trap(trapBase, trapEffect, trapDamage);

			Approvals.Verify(trap);
		}

		[Test]
		public void RandomFactoryTest()
		{
			DiceCup.SetRandom(new Random(100));

			var architect = new TrapArchitect();
			var effectList = new List<string>();
			int repeat = 100;
			for (int i = 0; i < repeat; i++)
			{
				effectList.Add(architect.GetTrapEffectFactory());
			}

			Approvals.VerifyAll(effectList, "Traps");
		}

		[Test]
		public void RandomTrapConstructionTest()
		{
			DiceCup.SetRandom(new Random(200));

			var architect = new TrapArchitect();
			var traps = new List<string>();
			var iterator = 100;
			for (int i = 0; i < iterator; i++)
			{
				var sb = new StringBuilder();
				sb.AppendLine(architect.GetTrapBaseFactory());
				sb.AppendLine(architect.GetTrapEffectFactory());
				sb.AppendLine(architect.GetTrapDamageFactory());
				traps.Add(sb.ToString());
			}

			Approvals.VerifyAll(traps, "Traps");
		}

		[Test]
		public void TrapObjectTest()
		{
			DiceCup.SetRandom(new Random(300));
			var architect = new TrapArchitect();
			var traps = new List<Trap>();

			var iterator = 100;
			for (int i = 0; i < iterator; i++)
			{
				traps.Add(architect.GetTrap());
			}

			Approvals.VerifyAll(traps, "Trap");
		}
	}
}
