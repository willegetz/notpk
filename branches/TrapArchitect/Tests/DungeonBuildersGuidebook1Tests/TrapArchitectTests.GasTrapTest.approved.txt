Gas Trap![0] = The vegetation
Releases a gas: Obscuring
Inflicts 2d6 points of damage

Gas Trap![1] = The hook
Releases a gas: Weakening
Inflicts 3d6 points of damage

Gas Trap![2] = The organ
Releases a gas: Sleep Inducing
Inflicts 3d6 points of damage

Gas Trap![3] = The column
Releases a gas: Blinding
Inflicts 4d6 points of damage

Gas Trap![4] = The chair
Releases a gas: Obscuring
Inflicts 3d6 points of damage

Gas Trap![5] = The bed
Releases a gas: Sleep Inducing
Inflicts 3d6 points of damage

Gas Trap![6] = The rug
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![7] = The forcefield
Releases a gas: Corroding
Inflicts 3d6 points of damage

Gas Trap![8] = The door
Releases a gas: Poisonous
Inflicts 3d6 points of damage

Gas Trap![9] = The bed
Releases a gas: Nauseating
Inflicts 2d6 points of damage

Gas Trap![10] = The holy/unholy font
Releases a gas: Blinding
Inflicts 2d6 points of damage

Gas Trap![11] = The floor
Releases a gas: Blinding
Inflicts 3d6 points of damage

Gas Trap![12] = The pendulum
Releases a gas: Nauseating
Inflicts 2d6 points of damage

Gas Trap![13] = The hook
Releases a gas: Nauseating
Inflicts 3d6 points of damage

Gas Trap![14] = The barrel
Releases a gas: Weakening
Inflicts 2d6 points of damage

Gas Trap![15] = The manacles
Releases a gas: Terrifying
Inflicts 2d6 points of damage

Gas Trap![16] = The painting
Releases a gas: Obscuring
Inflicts 2d6 points of damage

Gas Trap![17] = The pews
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![18] = The elevator (Mechanical)
Releases a gas: Obscuring
Inflicts 2d6 points of damage

Gas Trap![19] = The window
Releases a gas: Blinding
Inflicts 2d6 points of damage

Gas Trap![20] = The magic circle
Releases a gas: Weakening
Inflicts 2d6 points of damage

Gas Trap![21] = The arch
Releases a gas: Poisonous
Inflicts 2d6 points of damage

Gas Trap![22] = The elevator (Magical)
Releases a gas: Weakening
Inflicts 2d6 points of damage

Gas Trap![23] = The chute
Releases a gas: Sleep Inducing
Inflicts 1d8 points of damage

Gas Trap![24] = The manacles
Releases a gas: Obscuring
Inflicts 1d8 points of damage

Gas Trap![25] = The trophy
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![26] = The organ
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![27] = The arch
Releases a gas: Corroding
Inflicts 3d6 points of damage

Gas Trap![28] = The catwalk
Releases a gas: Obscuring
Inflicts 2d6 points of damage

Gas Trap![29] = The jar
Releases a gas: Weakening
Inflicts 3d6 points of damage

Gas Trap![30] = The crate
Releases a gas: Sleep Inducing
Inflicts 3d6 points of damage

Gas Trap![31] = The concealed door
Releases a gas: Terrifying
Inflicts 2d6 points of damage

Gas Trap![32] = The fresco
Releases a gas: Corroding
Inflicts 3d6 points of damage

Gas Trap![33] = The catwalk
Releases a gas: Blinding
Inflicts 2d6 points of damage

Gas Trap![34] = The bridge
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![35] = The basin
Releases a gas: Sleep Inducing
Inflicts 1d8 points of damage

Gas Trap![36] = The urn
Releases a gas: Corroding
Inflicts 10d6 points of damage

Gas Trap![37] = The table
Releases a gas: Nauseating
Inflicts 2d6 points of damage

Gas Trap![38] = The pillar
Releases a gas: Terrifying
Inflicts 2d6 points of damage

Gas Trap![39] = The barrel
Releases a gas: Nauseating
Inflicts 2d6 points of damage

Gas Trap![40] = The hourglass
Releases a gas: Sleep Inducing
Inflicts 2d6 points of damage

Gas Trap![41] = The button/switch
Releases a gas: Terrifying
Inflicts 10d10points of damage

Gas Trap![42] = The idol
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![43] = The chest
Releases a gas: Nauseating
Inflicts 3d6 points of damage

Gas Trap![44] = The portcullis
Releases a gas: Terrifying
Inflicts 1d4 points of damage

Gas Trap![45] = The stairway
Releases a gas: Poisonous
Inflicts 3d6 points of damage

Gas Trap![46] = The manacles
Releases a gas: Blinding
Inflicts 2d6 points of damage

Gas Trap![47] = The aquarium
Releases a gas: Poisonous
Inflicts 3d6 points of damage

Gas Trap![48] = The idol
Releases a gas: Obscuring
Inflicts 2d6 points of damage

Gas Trap![49] = The idol
Releases a gas: Poisonous
Inflicts 2d6 points of damage

Gas Trap![50] = The mosaic
Releases a gas: Terrifying
Inflicts 2d6 points of damage

Gas Trap![51] = The cask
Releases a gas: Sleep Inducing
Inflicts 3d6 points of damage

Gas Trap![52] = The furnace
Releases a gas: Terrifying
Inflicts 2d6 points of damage

Gas Trap![53] = The holy/unholy font
Releases a gas: Terrifying
Inflicts 1d8 points of damage

Gas Trap![54] = The illusion
Releases a gas: Sleep Inducing
Inflicts 2d6 points of damage

Gas Trap![55] = The bell
Releases a gas: Obscuring
Inflicts 4d6 points of damage

Gas Trap![56] = The barrel
Releases a gas: Poisonous
Inflicts 6d6 points of damage

Gas Trap![57] = The column
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![58] = The candelabra
Releases a gas: Poisonous
Inflicts 2d6 points of damage

Gas Trap![59] = The candelabra
Releases a gas: Obscuring
Inflicts 4d6 points of damage

Gas Trap![60] = The caldron
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![61] = The statue
Releases a gas: Obscuring
Inflicts 1d8 points of damage

Gas Trap![62] = The pews
Releases a gas: Weakening
Inflicts 2d6 points of damage

Gas Trap![63] = The rug
Releases a gas: Obscuring
Inflicts 2d6 points of damage

Gas Trap![64] = The crate
Releases a gas: Sleep Inducing
Inflicts 3d6 points of damage

Gas Trap![65] = The trunk
Releases a gas: Blinding
Inflicts 2d6 points of damage

Gas Trap![66] = The ladder
Releases a gas: Obscuring
Inflicts 3d6 points of damage

Gas Trap![67] = The telescope (Mechanical)
Releases a gas: Poisonous
Inflicts 6d6 points of damage

Gas Trap![68] = The magic circle
Releases a gas: Nauseating
Inflicts 2d6 points of damage

Gas Trap![69] = The bridge
Releases a gas: Obscuring
Inflicts 4d6 points of damage

Gas Trap![70] = The furnace
Releases a gas: Poisonous
Inflicts 6d6 points of damage

Gas Trap![71] = The hourglass
Releases a gas: Sleep Inducing
Inflicts 2d6 points of damage

Gas Trap![72] = The secret door
Releases a gas: Obscuring
Inflicts 3d6 points of damage

Gas Trap![73] = The wall
Releases a gas: Blinding
Inflicts 8d6 points of damage

Gas Trap![74] = The vase
Releases a gas: Terrifying
Inflicts 2d6 points of damage

Gas Trap![75] = The concealed door
Releases a gas: Blinding
Inflicts 6d6 points of damage

Gas Trap![76] = The illusion
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![77] = The statue
Releases a gas: Blinding
Inflicts 2d6 points of damage

Gas Trap![78] = The basin
Releases a gas: Obscuring
Inflicts 3d6 points of damage

Gas Trap![79] = The window
Releases a gas: Poisonous
Inflicts 1d4 points of damage

Gas Trap![80] = The barrel
Releases a gas: Sleep Inducing
Inflicts 2d6 points of damage

Gas Trap![81] = The brazier
Releases a gas: Blinding
Inflicts 3d6 points of damage

Gas Trap![82] = The fountain
Releases a gas: Blinding
Inflicts 2d6 points of damage

Gas Trap![83] = The bridge
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![84] = The table
Releases a gas: Obscuring
Inflicts 3d6 points of damage

Gas Trap![85] = The crystal ball
Releases a gas: Weakening
Inflicts 8d6 points of damage

Gas Trap![86] = The pit
Releases a gas: Blinding
Inflicts 2d6 points of damage

Gas Trap![87] = The railing
Releases a gas: Blinding
Inflicts 10d10points of damage

Gas Trap![88] = The pool
Releases a gas: Nauseating
Inflicts 2d6 points of damage

Gas Trap![89] = The armor
Releases a gas: Terrifying
Inflicts 3d6 points of damage

Gas Trap![90] = The table
Releases a gas: Terrifying
Inflicts 3d6 points of damage

Gas Trap![91] = The illusion
Releases a gas: Corroding
Inflicts 3d6 points of damage

Gas Trap![92] = The catwalk
Releases a gas: Nauseating
Inflicts 2d6 points of damage

Gas Trap![93] = The escalator (Magical)
Releases a gas: Weakening
Inflicts 2d6 points of damage

Gas Trap![94] = The chandelier
Releases a gas: Nauseating
Inflicts 2d6 points of damage

Gas Trap![95] = The ceiling
Releases a gas: Obscuring
Inflicts 2d6 points of damage

Gas Trap![96] = The bed
Releases a gas: Weakening
Inflicts 3d6 points of damage

Gas Trap![97] = The holy/unholy font
Releases a gas: Poisonous
Inflicts 2d6 points of damage

Gas Trap![98] = The crystal ball
Releases a gas: Corroding
Inflicts 2d6 points of damage

Gas Trap![99] = The mirror
Releases a gas: Sleep Inducing
Inflicts 4d6 points of damage

